cloudify/workflows/local.py

```diff
@@ -675,11 +686,15 @@ def _store_instance(self, node_instance, lock=True):
     def _instance_path(self, node_instance_id):
         return os.path.join(self._instances_dir, node_instance_id)
 
-    def get_node_instances(self, node_id=None):
+    def get_node_instances(self, node_id=None, evaluate_functions=True):
         instances = [self._get_node_instance(instance_id)
                      for instance_id in self._instance_ids()]
         if node_id:
             instances = [i for i in instances if i.node_id == node_id]
+        if evaluate_functions:
+            for instance in instances:
+                dsl_functions.evaluate_node_instance_functions(
+                    instance, self)
```

Comment:
There seems to be a lot of code duplication between FileStorage and InMemoryStorage. Is this something that can reasonably be refactored or would that be massively more hassle than it's worth at the moment?

Reply:
not sure about a lot - changed some things, lost 3? lines...